appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java

```diff
@@ -55,11 +56,16 @@ public String getName() {
     }
 
     @ReactMethod
-    public void read(String documentId, String partition, final Promise promise) {
-        Data.read(documentId, JsonElement.class, partition).thenAccept(new AppCenterConsumer<DocumentWrapper<JsonElement>>() {
+    public void read(String documentId, String partition, int ttl, final Promise promise) {
+        Data.read(documentId, JsonElement.class, partition, new ReadOptions(ttl)).thenAccept(new AppCenterConsumer<DocumentWrapper<JsonElement>>() {
 
             @Override
             public void accept(DocumentWrapper<JsonElement> documentWrapper) {
+                if (documentWrapper.getError() != null) {
+                    DataException dataException = documentWrapper.getError();
+                    promise.reject("Read failed", dataException);
```

Comment:
Did you test how the exception looks like in Javascript? What properties it contains?

Reply:
Something like this for an invalid documentId:

<img width="763" alt="Screen Shot 2019-06-21 at 3 21 11 PM" src="https://user-images.githubusercontent.com/8866892/59954175-4187a980-9438-11e9-89aa-d4e02d2ca299.png">